step.go

```diff
@@ -35,15 +42,31 @@ func (a ArtifactPull) ProcessConfig() (Config, error) {
 
 	stepconf.Print(input)
 
+	finishedStages := a.envRepository.Get("FINISHED_STAGES")
+
+	var finishedStagesModel model.FinishedStages
+	if err := json.Unmarshal([]byte(finishedStages), &finishedStagesModel); err != nil {
+		return Config{}, fmt.Errorf("failed to parse step inputs: %w", err)
+	}
+
 	// TODO: validate inputs here and possibly convert from string to a concrete type
 	//nolint:gosimple
 	return Config{
-		Verbose: input.Verbose,
+		Verbose:         input.Verbose,
+		ArtifactSources: strings.Split(input.ArtifactSources, ","),
+		FinishedStages:  finishedStagesModel,
 	}, nil
 }
 
 func (a ArtifactPull) Run(cfg Config) (Result, error) {
 	a.logger.EnableDebugLog(cfg.Verbose)
+	buildIdGetter := NewDefaultBuildIDGetter(cfg.FinishedStages, cfg.ArtifactSources)
+	buildIDs, err := buildIdGetter.GetBuildIDs()
+	if err != nil {
+		return Result{}, err
+	}
+
+	a.logger.Printf("%+v", buildIDs)
```

Comment:
Do we always want to print these out?

Reply:
No, I just would like to mark the code. I left the logging but I inserted an extra `TODO` comment.